executor/importer/table_import.go

```diff
@@ -0,0 +1,402 @@
+	// todo: add job id too
+	sortPathSuffix := "import-" + strconv.Itoa(int(tidbCfg.Port))
+	sortPath := filepath.Join(tidbCfg.TempDir, sortPathSuffix)
+
+	if info, err := os.Stat(sortPath); err != nil {
+		if !os.IsNotExist(err) {
+			e.logger.Error("stat sort dir failed", zap.String("path", sortPath), zap.Error(err))
+			return "", errors.Trace(err)
+		}
+	} else if info.IsDir() {
+		// Currently remove all dir to clean garbage data.
+		// TODO: when do checkpoint should change follow logic.
+		err := os.RemoveAll(sortPath)
+		if err != nil {
+			e.logger.Error("remove sort dir failed", zap.String("path", sortPath), zap.Error(err))
+		}
+	}
+
+	err := os.MkdirAll(sortPath, 0o700)
+	if err != nil {
+		e.logger.Error("failed to make sort dir", zap.String("path", sortPath), zap.Error(err))
+		return "", errors.Trace(err)
+	}
+	e.logger.Info("sort dir prepared", zap.String("path", sortPath))
+	return sortPath, nil
+}
+
+func newTableImporter(ctx context.Context, e *LoadDataController) (ti *tableImporter, err error) {
+	idAlloc := kv.NewPanickingAllocators(0)
+	tbl, err := tables.TableFromMeta(idAlloc, e.Table.Meta())
+	if err != nil {
+		return nil, errors.Annotatef(err, "failed to tables.TableFromMeta %s", e.Table.Meta().Name)
+	}
+
+	tidbCfg := tidb.GetGlobalConfig()
+	dir, err := prepareSortDir(e)
+	if err != nil {
+		return nil, err
+	}
+
+	hostPort := net.JoinHostPort("127.0.0.1", strconv.Itoa(int(tidbCfg.Status.StatusPort)))
+	tls, err := common.NewTLS(
+		tidbCfg.Security.ClusterSSLCA,
+		tidbCfg.Security.ClusterSSLCert,
+		tidbCfg.Security.ClusterSSLKey,
+		hostPort,
+		nil, nil, nil,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// Disable GC because TiDB enables GC already.
+	keySpaceName := tidb.GetGlobalKeyspaceName()
+	// the kv store we get is a cached store, so we can't close it.
+	kvStore, err := GetKVStore(fmt.Sprintf("tikv://%s?disableGC=true&keyspaceName=%s", tidbCfg.Path, keySpaceName), tls.ToTiKVSecurityConfig())
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	backendConfig := local.BackendConfig{
+		PDAddr:                  tidbCfg.Path,
+		LocalStoreDir:           dir,
+		MaxConnPerStore:         config.DefaultRangeConcurrency,
+		ConnCompressType:        config.CompressionNone,
+		WorkerConcurrency:       config.DefaultRangeConcurrency * 2,
+		KVWriteBatchSize:        config.KVWriteBatchSize,
+		CheckpointEnabled:       true,
+		MemTableSize:            int(config.DefaultEngineMemCacheSize),
+		LocalWriterMemCacheSize: int64(config.DefaultLocalWriterMemCacheSize),
+		ShouldCheckTiKV:         true,
+		DupeDetectEnabled:       false,
+		DuplicateDetectOpt:      local.DupDetectOpt{ReportErrOnDup: false},
+		StoreWriteBWLimit:       0,
+		ShouldCheckWriteStall:   false,
+		MaxOpenFiles:            int(util.GenRLimit()),
+		KeyspaceName:            keySpaceName,
+	}
+
+	tableMeta := &mydump.MDTableMeta{
+		DB:        e.DBName,
+		Name:      e.Table.Meta().Name.O,
+		DataFiles: e.toMyDumpFiles(),
```

Comment:
Add TODO for `IsRowOrdered`?

Reply:
https://github.com/pingcap/tidb/pull/42817/commits/6ff452769d65ea6b584edb2e1294fbd14e05862e